### PR TITLE
Feature/task policies

### PR DESCRIPTION
**Pull request by NikitaMasalkov**

### CODE REVIEW

File: app/policies/task_policy.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class TaskPolicy < ApplicationPolicy
+
+  def creator?
+    @user == @record.creator
```
Comment (vkalach) [inline app/policies/task_policy.rb]: compare by id?

File: app/views/tasks/_task_form.html.erb [hunk 2]
```diff
@@ -1,31 +1,41 @@
...
+            <%= form.text_field :title, class:'form-control', required: true %>
+          </div>
+          <div class="form-group">
+            <%= form.label :description, class: 'control-label' %>
+            <%= form.text_area :description, class: 'form-control' %>
+          </div>
+          <div class="form-group">
+            <%= form.label :users, class: 'control-label' %>
+            <% if is_new %>
+              <%= form.select :user_ids, users.map{ |user| [user.username, user.id] }, {}, { class: 'form-control', multiple: true, size: users.length } %>
```
Comment (vkalach) [inline app/views/tasks/_task_form.html.erb]: spaces

File: app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -18,13 +26,21 @@ def create
   def edit
     task = find_task
     users = User.all
+    assigned_users_ids = task.assigned_users.map(&:id)
     statuses = Task.statuses
-    render locals: { task: task, users: users, statuses: statuses }
+    is_new = false
```
Comment (vkalach) [inline app/controllers/tasks_controller.rb]: find another way to check if the record is new or not

File: app/controllers/tasks_controller.rb [hunk 4]
```diff
@@ -1,14 +1,22 @@
 class TasksController < ApplicationController
+
+  before_action :user_presence, except: :show
```
Comment (vkalach) [inline app/controllers/tasks_controller.rb]: use devise callback for this

File: app/controllers/tasks_controller.rb [hunk 5]
```diff
@@ -1,14 +1,22 @@
...
+    render locals: { task: task,
+                     users: users,
+                     statuses: statuses,
+                     is_new: is_new }
   end
 
   def create
     task = Task.new(task_params)
     if task.save
+      task.add_creator(current_user)
```
Comment (vkalach) [inline app/controllers/tasks_controller.rb]: add creator in a different way (place)

File: app/models/task.rb [hunk 6]
```diff
@@ -11,6 +12,29 @@ class Task < ApplicationRecord
...
 
+  def add_creator(user)
+    users << user
+    assignment = Assignment.find_by(user_id: user.id, task_id: id)
+    assignment.creator = true
+    assignment.assigned = false
+    assignment.save
+  end
+
+  def add_assignee(user)
```
Comment (vkalach) [inline app/models/task.rb]: did you add this just for seeds? :smiley:

File: app/views/dashboards/index.html.erb [hunk 7]
```diff
@@ -35,23 +35,27 @@
   </div>
 </div>
 
-<div class="container pt-3 text-center">
-  <div class="row">
-    <div class="col-md-6">
-      <%= link_to 'Create new task', new_task_path, class: "btn btn-primary"%>
+<% if current_user.present? %>
```
Comment (vkalach) [inline app/views/dashboards/index.html.erb]: use pundit policy

File: app/views/tasks/show.html.erb [hunk 8]
```diff
@@ -4,11 +4,13 @@
       Task: <span class="text-success"> <%= task.title %></span>
     </div>
   </div>
-  <div class="row">
-    <div class=" col-md-12 text-center display-2" >
-      <%= link_to "Edit this task", edit_task_path(task), class: "btn btn-primary" %>
+  <% if policy(task).creator? || policy(task).assigned? %>
```
Comment (vkalach) [inline app/views/tasks/show.html.erb]: use another policy

File: app/models/task.rb (hunk 6 above)
Comment (NikitaMasalkov) [reply app/models/task.rb]: yep :)

File: app/controllers/tasks_controller.rb [hunk 9]
```diff
@@ -1,13 +1,18 @@
...
     statuses = Task.statuses
-    render locals: { task: task, users: users, statuses: statuses }
+    render locals: { task: task,
+                     users: users,
+                     statuses: statuses }
   end
 
   def create
     task = Task.new(task_params)
+    task.set_creator(current_user)
```
Comment (NikitaMasalkov) [inline app/controllers/tasks_controller.rb]: to be fixed when moving to form object

**Pull request merged**